src/wp-includes/block-template-utils.php

```diff
@@ -902,12 +902,32 @@ function block_footer_area() {
 	block_template_part( 'footer' );
 }
 
+/**
+ * Filters theme directories that should be ignored during export.
+ *
+ * @since 6.0.0
+ *
+ * @param string $path The path of the file in the theme.
+ * @return Bool Whether this file is in an ignored directory.
+ */
+function wp_is_theme_directory_ignored( $path ) {
+	$directories_to_ignore = array( '.git', 'node_modules', 'vendor' );
```

Comment:
Do we want to look for other VCS'? We check for `array( '.svn', '.git', '.hg', '.bzr' )` in `WP_Automatic_Updater`.

Reply:
If we're now including the whole theme in the export, excluding `vendor` may be incorrect. It isn't always/only for development dependencies. So excluding it here would preclude using runtime composer dependencies.

This is something the WP Theme Review Team is [doing](https://github.com/WPTT/control-color-alpha), though maybe this would never happen for FSE themes.

Reply:
> Do we want to look for other VCS'? We check for array( '.svn', '.git', '.hg', '.bzr' ) in WP_Automatic_Updater.

Done

Reply:
> maybe this would never happen for FSE themes.

This is my hunch but I'm happy to remove it. The main motivation here is that TT2 has a vendor folder which is very large (30MB) so if people are using that as a base theme (which seems fairly likely), they will end up copying that every time. Happy to remove it though if you prefer.

Reply:
Yeah I definitely think we don't want to be zipping up 30MB here. I think your intuition is probably the best way to go. Maybe make the ignore directories list filterable?